olp-cpp-sdk-dataservice-read/src/repositories/PartitionsCacheRepository.cpp

```diff
@@ -60,6 +60,13 @@ std::string CreateKey(const std::string& hrn, const std::string& layer_id,
 std::string CreateKey(const std::string& hrn, const int64_t catalogVersion) {
   return hrn + "::" + std::to_string(catalogVersion) + "::layerVersions";
 }
+std::string CreateKey(const std::string& hrn, const std::string& layer,
+                      olp::geo::TileKey key, int32_t depth,
+                      const boost::optional<int64_t>& version) {
+  return hrn + "::" + key.ToHereTile() + "::" + layer +
```

Comment:
Why are you putting the TileKey in second place? If we want to delete the entire data for a layer will this work? No. Keep the same structure! 
Which is in this case:
`<hrn>::<layer>::<tilekey>::<version>::<depth>::quadtree`